general/array.hpp

```diff
@@ -23,6 +23,7 @@
 #include <cstring>
 #include <algorithm>
 #include <type_traits>
+#include <initializer_list>
```

Comment:
minor: `#include <type_traits>` due to `std::enable_if` and `std::initializer_list` ?

Reply:
It's already included in this header, but I added the include to `vector.hpp` in c232029d84270e0ec0fee02aea5efb7e3719ff7d.